toolchain/sem_ir/function.h

```diff
@@ -49,6 +50,11 @@ struct FunctionFields {
   // function, in lexical order. The first block is the entry block. This will
   // be empty for declarations that don't have a visible definition.
   llvm::SmallVector<InstBlockId> body_block_ids = {};
+
+  // If the function is imported from C++, points to the Clang declaration in
+  // the AST. Used for mangling. The AST is owned by `CompileSubcommand` so we
+  // expect it to be live from `Function` creation to mangling.
+  const clang::NamedDecl* cpp_clang_decl = nullptr;
```

Comment:
The SemIR is intended to be stored on disk, which would invalidate this pointer. Is there an easy way to get a more durable reference? Can you add a TODO to take another look at ensuring this can be easily serialized?

```suggestion
  // TODO: Ensure we can easily serialize/deserialize this.
  const clang::NamedDecl* cpp_clang_decl = nullptr;
```

Reply:
Added a TODO.
Is this also true for [`File::cpp_ast_`](https://github.com/carbon-language/carbon-lang/blob/92b3e61289ee1b8e3c08dc45418add1b00081aea/toolchain/sem_ir/file.h#L311) or is that use case different?
Do we have any serialization / deserialization tests that I can try?

Reply:
I assumed the AST pointer would be more straightforward to replace (i.e., it becomes some reference to whatever serialized state). The decl ID approach Richard brought up today might be worth checking (in particular if it's 32-bit, it'd be smaller than a pointer).

Serialization/deserialization is an open question. Note similarly, ImportIR carries pointers to other files. A pointer isn't infeasible on the whole, I'm just wanting to be careful we don't get into a corner.

Reply:
SG, added the TODO for now.